util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/rule/TestElasticsearchClient.java

```diff
@@ -107,12 +109,23 @@ public boolean exists() {
 		}
 
 		public IndexClient deleteAndCreate() {
-			TestElasticsearchClient.this.deleteAndCreateIndex( primaryIndexName, writeAlias, readAlias );
+			return deleteAndCreate( true );
+		}
+
+		public IndexClient deleteAndCreate(boolean withMaxResultWindow) {
+			TestElasticsearchClient.this.deleteAndCreateIndex( primaryIndexName, writeAlias, readAlias,
+					withMaxResultWindow );
 			return this;
 		}
 
 		public IndexClient deleteAndCreate(String settingsPath, String settings) {
-			JsonObject settingsAsJsonObject = buildStructuredSettings( settingsPath, settings );
+			// by default we add the withMaxResultWindow,
+			// so that we don't have any validation errors for this attribute
```

Comment:
Again, we shouldn't have validation errors when the max result window is the default. The validation code should assume that `null` means `10,000`. I left a comment on the validation code with a solution.